src/renderers/dom/fiber/ReactDOMFiberEntry.js

```diff
@@ -531,6 +531,25 @@ function renderSubtreeIntoContainer(
   );
 
   if (__DEV__) {
+    if (container._reactRootContainer && container.nodeType !== COMMENT_NODE) {
+      const hostInstance = DOMRenderer.findHostInstance(
+        container._reactRootContainer.current,
+      );
```

Comment:
Does it help that we now do the mutating inside of `ReactFiberCommitWork`? That is where we do other DOM updates.

If we really want to avoid adding this flag then I can implement another method on `DOMRenderer` but I think @sebmarkbage's point that "It's a bit awkward that we expose a new method on the public Reconciler API that is only ever needed for a specific DEV warning." is valid, and I'd rather avoid that.